instrumentation/jvm-metrics/src/main/java/com/splunk/opentelemetry/instrumentation/jvmmetrics/AllocatedMemoryMetrics.java

```diff
@@ -17,42 +17,30 @@
 package com.splunk.opentelemetry.instrumentation.jvmmetrics;
 
 import com.sun.management.ThreadMXBean;
-import io.micrometer.core.instrument.FunctionCounter;
-import io.micrometer.core.instrument.MeterRegistry;
-import io.micrometer.core.instrument.binder.BaseUnits;
-import io.micrometer.core.instrument.binder.MeterBinder;
 import java.lang.management.ManagementFactory;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
-public class AllocatedMemoryMetrics implements MeterBinder {
+public class AllocatedMemoryMetrics {
   public static final String METRIC_NAME = "process.runtime.jvm.memory.allocated";
   private final boolean hasComSunThreadMXBean = hasComSunThreadMXBean();
+  private final AllocationTracker allocationTracker = createAllocationTracker();
 
-  @Override
-  public void bindTo(MeterRegistry registry) {
-    if (!hasComSunThreadMXBean) {
-      return;
-    }
+  public boolean isAvailable() {
+    return allocationTracker != null;
+  }
 
-    if (!isThreadAllocatedMemoryEnabled()) {
-      return;
-    }
+  public long getCumulativeAllocationTotal() {
+    return allocationTracker.getCumulativeAllocationTotal();
+  }
 
-    AllocationTracker allocationTracker = new AllocationTracker();
-    // FunctionCounter keeps a weak reference to allocationTracker. To ensure it is not collected
-    // we pass a capturing lambda as the function instead of method reference
-    // AllocationTracker::getCumulativeAllocationTotal
-    FunctionCounter.builder(
-            METRIC_NAME,
-            allocationTracker,
-            (unused) -> allocationTracker.getCumulativeAllocationTotal())
-        .description("Approximate sum of heap allocations")
-        .baseUnit(BaseUnits.BYTES)
-        .tag("type", "heap")
-        .register(registry);
+  private AllocationTracker createAllocationTracker() {
+    if (hasComSunThreadMXBean && isThreadAllocatedMemoryEnabled()) {
```

Comment:
This is confusing now because it relies on the `hasComSunThreadMXBean` field to have been assigned before we get here, which now relies on declaration ordering, which is fragile. Better to remove the `hasComSunThreadMXBean` field altogether and just invoke the method.

Reply:
`hasComSunThreadMXBean` is a final field, I changed it also to be static. Is this enough?

Reply:
Sure. I think it's less readable than calling it inline because the reader has to understand when/where that field was assigned, but it's fine.